src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (this.length === 1) {
+      return this[0].toString();
+    }
```

Comment:
Why do you need this ?

Reply:
cos no need in anything if this.length === 1

Reply:
this if statement makes no sense, because if the length of this gonna be 1, so you would return that without separatator because of that rule.  
```
if (i < this.length - 1) {
       result += separator;
     }
```

Reply:
but no need in cycle, no need in let result, no need in anything else, cos i can return what i search immiidiatly